acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/Eth2CustomNetworkFileAcceptanceTest.java

```diff
@@ -0,0 +1,84 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.tests.signing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import tech.pegasys.teku.bls.BLS;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSSecretKey;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.Eth2SigningRequestBody;
+import tech.pegasys.web3signer.dsl.utils.Eth2BlockSigningRequestUtil;
+import tech.pegasys.web3signer.dsl.utils.MetadataFileHelpers;
+import tech.pegasys.web3signer.signing.KeyType;
+
+import java.nio.file.Path;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.io.Resources;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class Eth2CustomNetworkFileAcceptanceTest extends SigningAcceptanceTestBase {
+  private static final String PRIVATE_KEY =
+      "3ee2224386c82ffea477e2adf28a2929f5c349165a4196158c7f3a2ecca40f35";
+
+  private static final MetadataFileHelpers metadataFileHelpers = new MetadataFileHelpers();
+  private static final BLSSecretKey key =
+      BLSSecretKey.fromBytes(Bytes32.fromHexString(PRIVATE_KEY));
+  private static final BLSKeyPair keyPair = new BLSKeyPair(key);
+  private static final BLSPublicKey publicKey = keyPair.getPublicKey();
+
+  @BeforeEach
+  void setup() {
+    final String configFilename = publicKey.toString().substring(2);
+    final Path keyConfigFile = testDirectory.resolve(configFilename + ".yaml");
+    metadataFileHelpers.createUnencryptedYamlFileAt(keyConfigFile, PRIVATE_KEY, KeyType.BLS);
+  }
+
+  @Test
+  void signAndVerifyBlockV2Signature() throws Exception {
+    final Path networkConfig = Path.of(Resources.getResource("eth2/network_config.yaml").getPath());
+    setupEth2SignerWithCustomNetworkConfig(networkConfig.toString());
+
+    final Spec spec = SpecFactory.create(networkConfig.toString());
+
+    // network_config.yaml: slots per epoch: 6, Altair fork epoch: 6, Bellatrix fork epoch: 8
+    for (int forkEpoch = 4; forkEpoch < 10; forkEpoch++) {
+      final int startingSlot = forkEpoch * 6;
+      for (int slot = startingSlot; slot < startingSlot + 6; slot++) {
```

Comment:
Some magic values here...would it make sense to replace 6 with `spec.getGenesisSpecConfig().getSlotsPerEpoch()`?

Is the intent behind epoch 4 and 10 to cover the range that includes the forks at 6 and 8?
If so, could this be 5 and 9 to make the range and therefore number of asserts smaller?

The comment helped but might be even clearer to associate the range with the config values. Not sure how to get hold of this data dynamically, but maybe they could at least be constants with the same name as the config in network_config.yaml?
e.g. ALTAIR_FORK_EPOCH - 1 and BELLATRIX_FORK_EPOCH + 1

Reply:
Also, what's the value in asserting every slot in the epoch? I'm assuming there is one, I'd just like to understand :)

Reply:
thanks for the hint, I've simplified the logic. The main idea was to cover all three forks with valid range of slots.